include/api.php

```diff
@@ -3664,6 +3667,336 @@ function api_friendica_notification_seen($type){
+		$result = q("DELETE FROM `mail` WHERE `uid` = %d AND `id` = %d AND `parent-uri` = '%s'", 
+			intval($uid), 
+			intval($id), 
+			dbesc($parenturi));
+
+		if ($result) {
+			// return success
+			$answer = array('result' => 'ok', 'message' => 'message deleted');
+			return api_format_data("direct_message_delete", $type, array('$result' => $answer));
+		}
+		else {
+			$answer = array('result' => 'error', 'message' => 'unknown error');
+			return api_format_data("direct_messages_delete", $type, array('$result' => $answer));
+		}
+	}
+	api_register_func('api/friendica/direct_messages_delete', 'api_friendica_direct_messages_delete', true);
+
+
+	/**
+	 * @brief search for direct_messages containing a searchstring through api
+	 *
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_friendica_direct_messages_search($type){
+		$a = get_app();
+
+		if (api_user()===false) throw new ForbiddenException();
+
+		// params
+		$user_info = api_get_user($a);
+		$searchstring = (x($_REQUEST,'searchstring') ? $_REQUEST['searchstring'] : "");
+		$uid = $user_info['uid'];
+	
+		// error if no searchstring specified
+		if ($searchstring == "") {
+			$answer = array('result' => 'error', 'message' => 'searchstring not specified');
+			return api_format_data("direct_messages_search", $type, array('$result' => $answer));
+		}
+
+		// get data for the specified searchstring
+		$r = q("SELECT `mail`.*, `contact`.`nurl` AS `contact-url` FROM `mail`,`contact` WHERE `mail`.`contact-id` = `contact`.`id` AND `mail`.`uid`=%d AND `body` LIKE '%s' ORDER BY `mail`.`id` DESC",
+			intval($uid),
+			dbesc('%'.$searchstring.'%')
+		);
+
+		$profile_url = $user_info["url"];
+		// message if nothing was found
+		if (count($r) == 0) 
+			$success = array('success' => false, 'search_results' => 'nothing found');
+		else {
+			$ret = Array();
+			foreach($r as $item) {
+				if ($box == "inbox" || $item['from-url'] != $profile_url){
+					$recipient = $user_info;
+					$sender = api_get_user($a,normalise_link($item['contact-url']));
+				}
+				elseif ($box == "sentbox" || $item['from-url'] == $profile_url){
+					$recipient = api_get_user($a,normalise_link($item['contact-url']));
+					$sender = $user_info;
+				}
+				$ret[]=api_format_messages($item, $recipient, $sender);
+			}
+			$success = array('success' => true, 'search_results' => $ret);
+		}
+
+		return api_format_data("direct_message_search", $type, array('$result' => $success));
+	}
+	api_register_func('api/friendica/direct_messages_search', 'api_friendica_direct_messages_search', true);
+
+
+	/**
+	 * @brief returns all messages for a specified parenturi, similar to api/direct_messages/conversation but enhanced to return parenturi and seen state
+	 *
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_friendica_direct_messages_conversation($type) {
```

Comment:
Can't this function be replaced by the existing function with maybe an additional parameter of the difference to the existing function is more than the returned values?